### PR TITLE
epic(esl-select): reworked using esl-popup (#1321)

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #1321
Closes: #1421

### CODE REVIEW

**Comment by ala-n** [comment]
Accepted for nearest versions

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-forms/esl-select/core/esl-select-dropdown.ts [hunk 1]
```diff
@@ -86,21 +87,10 @@ export class ESLSelectDropdown extends ESLToggleable {
...
-  public updatePosition(): void {
-    if (!this.activator) return;
-    const windowY = window.scrollY || window.pageYOffset;
-    const rect = this.activator.getBoundingClientRect();
-
-    this.style.top = `${windowY + rect.top + rect.height}px`;
-    this.style.left = `${rect.left}px`;
-    this.style.width = `${rect.width}px`;
+  protected override _updatePosition(): void {
+    const select = this.activator;
```
Comment (julia-murashko) [inline src/modules/esl-forms/esl-select/core/esl-select-dropdown.ts:91]: why don't we use $ to name the variable 'select'?

**Review by julia-murashko — APPROVED** [review]

File: src/modules/esl-forms/esl-select/core/esl-select-renderer.ts [hunk 2]
```diff
@@ -21,28 +21,41 @@ export class ESLSelectRenderer extends ESLBaseElement {
...
-  constructor() {
-    super();
+  /** Inner remove button */
+  @memoize()
+  protected get $remove(): HTMLElement {
+    const $remove = document.createElement('button');
+    $remove.type = 'button';
+    $remove.setAttribute('aria-label', 'Clear');
+    $remove.classList.add('esl-select-clear-btn');
+    $remove.classList.add('icon-nav-close-menu');
```
Comment (dshovchko) [inline src/modules/esl-forms/esl-select/core/esl-select-renderer.ts:41]:
```suggestion
    $remove.className = 'esl-select-clear-btn icon-nav-close-menu';
```
Its shorter. Besides, className is much faster.

**Review by dshovchko — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**